packages/x-data-grid/src/hooks/features/dataSource/useGridDataSourceBase.ts

```diff
@@ -92,7 +92,7 @@ export const useGridDataSourceBase = <Api extends GridPrivateApiCommunity>(
   );
 
   const fetchRows = React.useCallback<GridDataSourceApiBase['fetchRows']>(
-    async (parentId, params) => {
+    async (parentId, params): Promise<void> => {
```

Comment:
```suggestion
    async (parentId, params) => {
```
useCallback is already typed